src/erlcloud_athena.erl

```diff
@@ -0,0 +1,436 @@
+%% '
+%%
+-spec list_query_executions() -> {ok, map()} | {error, any()}.
+list_query_executions() ->
+    list_query_executions(#{}, default_config()).
+
+-spec list_query_executions(map() | aws_config()) ->
+    {ok, map()} | {error, any()}.
+list_query_executions(Config) when is_record(Config, aws_config) ->
+    list_query_executions(#{}, Config);
+list_query_executions(PaginationMap) when is_map(PaginationMap) ->
+    list_query_executions(PaginationMap, default_config()).
+
+-spec list_query_executions(map(), aws_config()) ->
+    {ok, map()} | {error, any()}.
+list_query_executions(PaginationMap, Config) ->
+    request(Config, "ListQueryExecutions", PaginationMap).
+
+%% @doc
+%% Athena API:
+%% http://docs.aws.amazon.com/athena/latest/APIReference/API_StartQueryExecution.html
+%%
+%% `
+%% erlcloud_athena:start_query_execution(
+%%                              <<"some-client-request-token">>,
+%%                              <<"db-name">>,
+%%                              <<"select * from some-tbl">>,
+%%                              <<"s3://some-bucket">>,
+%%                              <<"SSE_KMS">>,
+%%                              <<"some-kms-key-id">>}]
+%% ).
+%% '
+%%
+-spec start_query_execution(binary(), binary(), binary(), binary()) ->
+    {ok, binary()} | {error, any()}.
+start_query_execution(ClientReqToken, Db, Query, OutputLocation) ->
+    start_query_execution(ClientReqToken, Db, Query, OutputLocation, undefined,
+                          undefined, default_config()).
+
+-spec start_query_execution(binary(), binary(), binary(), binary(),
+                            aws_config()) ->
+    {ok, binary()} | {error, any()}.
+start_query_execution(ClientReqToken, Db, Query, OutputLocation, Config)
+  when is_record(Config, aws_config) ->
+    start_query_execution(ClientReqToken, Db, Query, OutputLocation, undefined,
+                          undefined, Config).
+
+-spec start_query_execution(binary(), binary(), binary(), binary(),
+                            binary() | undefined,
+                            binary() | undefined) ->
+    {ok, binary()} | {error, any()}.
+start_query_execution(ClientReqToken, Db, Query, OutputLocation,
+                      EncryptionOption, KmsKey) ->
+    start_query_execution(ClientReqToken, Db, Query, OutputLocation,
+                          EncryptionOption, KmsKey, default_config()).
+
+-spec start_query_execution(binary(), binary(), binary(), binary(),
+                            binary() | undefined,
+                            binary() | undefined,
+                            aws_config()) ->
+    {ok, binary()} | {error, any()}.
+start_query_execution(ClientReqToken, Db, Query, OutputLocation,
+                      EncryptionOption, KmsKey, Config) ->
+    EncryptConfig = get_encrypt_config(EncryptionOption, KmsKey),
+    OutputConfig  = #{<<"OutputLocation">> => OutputLocation},
+    ResultConfig  = maps:merge(OutputConfig, EncryptConfig),
```

Comment:
stylish
why not `EncryptConfig#{<<"OutputLocation">> => OutputLocation}`
and same for all other maps:merge()

Reply:
ah gosh, i thought i got rid of this 'merges' everywhere. Just missed those